hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bloom/HoodieBaseBloomIndexCheckFunction.java

```diff
@@ -33,44 +33,42 @@
 import java.util.Iterator;
 import java.util.List;
 
-import scala.Tuple2;
-
 /**
  * Function performing actual checking of list containing (fileId, hoodieKeys) against the actual files.
  */
 //TODO we can move this class into the hudi-client-common and reuse it for spark client
-public class HoodieFlinkBloomIndexCheckFunction
-        implements Function<Iterator<Tuple2<String, HoodieKey>>, Iterator<List<KeyLookupResult>>> {
+public class HoodieBaseBloomIndexCheckFunction
+    implements Function<Iterator<Pair<String, HoodieKey>>, Iterator<List<KeyLookupResult>>> {
 
   private final HoodieTable hoodieTable;
 
   private final HoodieWriteConfig config;
 
-  public HoodieFlinkBloomIndexCheckFunction(HoodieTable hoodieTable, HoodieWriteConfig config) {
+  public HoodieBaseBloomIndexCheckFunction(HoodieTable hoodieTable, HoodieWriteConfig config) {
     this.hoodieTable = hoodieTable;
     this.config = config;
   }
 
   @Override
-  public Iterator<List<KeyLookupResult>> apply(Iterator<Tuple2<String, HoodieKey>> fileParitionRecordKeyTripletItr) {
+  public Iterator<List<KeyLookupResult>> apply(Iterator<Pair<String, HoodieKey>> fileParitionRecordKeyTripletItr) {
```

Comment:
any reason to replace Tuple2 with Pair?

Reply:
It seems better to move most implements from FlinkHoodieBloomIndex.java to HoodieBaseBloomIndex.java, then it can be shared with java client. Since java client not dependent on scala, so we need to replace scala.Tuple2 with Pair.